api/app/Models/User.php

```diff
@@ -159,4 +151,265 @@ public function getIsProfileCompleteAttribute(): bool
+    public function filterByJobLookingStatus(Builder $query, array $statuses): Builder
+    {
+        // JobLookingStatus acts as an OR filter. The query should return users with ANY of the statuses.
+        $query->where(function($query) use ($statuses) {
+            foreach($statuses as $index => $status) {
+                if ($index === 0) {
+                    // First iteration must use where instead of orWhere
+                    $query->where('job_looking_status', $status);
+                } else {
+                    $query->orWhere('job_looking_status', $status);
+                }
+            }
+        });
+        return $query;
+    }
+    public function filterBySkills(Builder $query, array $skills): Builder
+    {
+        if (empty($skills)) {
+            return $query;
+        }
+
+        // skills act as an AND filter. The query should only return candidates with ALL of the skills.
+        $query->whereExists(function ($query) use ($skills) {
+            $query->select(DB::raw('null'))
+            ->from(function ($query) {
+                $query->selectRaw('experiences.user_id, jsonb_agg(experience_skills.skill_id) as user_skills_grouped')
+                ->from('experience_skills')
+                ->joinSub(function ($query) {
+                    $query->select('award_experiences.id as experience_id', 'award_experiences.user_id')
+                    ->from('award_experiences')
+                    ->unionAll(function ($query) {
+                        $query->select('community_experiences.id as experience_id', 'community_experiences.user_id')
+                        ->from('community_experiences');
+                    })
+                    ->unionAll(function ($query) {
+                        $query->select('education_experiences.id as experience_id', 'education_experiences.user_id')
+                        ->from('education_experiences');
+                    })
+                    ->unionAll(function ($query) {
+                        $query->select('personal_experiences.id as experience_id', 'personal_experiences.user_id')
+                        ->from('personal_experiences');
+                    })
+                    ->unionAll(function ($query) {
+                        $query->select('work_experiences.id as experience_id', 'work_experiences.user_id')
+                        ->from('work_experiences');
+                    });
+                }, 'experiences', function ($join) {
+                    $join->on('experience_skills.experience_id', '=', 'experiences.experience_id');
+                })
+                ->groupBy('experiences.user_id');
+            }, "aggregate_experiences")
+            ->whereJsonContains('aggregate_experiences.user_skills_grouped', $skills)
+            ->whereColumn('aggregate_experiences.user_id', 'users.id');
+        });
+        return $query;
+    }
+    public function filterByClassifications(Builder $query, array $classifications): Builder
```

Comment:
There was a small (but important!) fix to the filterByClassifications in the CandidatePool yesterday in https://github.com/GCTC-NTGC/gc-digital-talent/pull/2805 .  Could you please backport that fix here?

Reply:
Will do.

Reply:
I'd actually already fixed this but I added the tests and changed my method to match the one in #2805 since it feels clearer anyway.